aspnetcore/signalr/authn-and-authz.md

```diff
@@ -176,6 +176,62 @@ public class ChatHub: Hub
 }
 ```
 
+::: moniker range=">= aspnetcore-3.0"
+
+### Use authorization handlers to customize hub method authorization
+
+SignalR provides a custom resource to authorization handlers when a hub method requires authorization. The resource is an instance of `HubInvocationContext`. The `HubInvocationContext` includes the `HubCallerContext`, the name of the hub method being invoked, and the arguments to the hub method.
+
+The following example shows a custom handler that only allows users with the same name as the hub method to invoke the method.
+
+```csharp
+public class HubMethodAuthorizationHandler :
+    AuthorizationHandler<HubMethodRequirement, HubInvocationContext>
+{
+    protected override Task HandleRequirementAsync(AuthorizationHandlerContext context,
+                                                   HubMethodRequirement requirement,
+                                                   HubInvocationContext resource)
+    {
+        if (context.User?.FindFirst(ClaimTypes.NameIdentifier)?.Value == resource.HubMethodName)
+        {
+            context.Succeed(requirement);
+        }
+
+        return Task.CompletedTask;
+    }
+}
+
+public class HubMethodRequirement : IAuthorizationRequirement { }
```

Comment:
This seems really weird. Should we be using OperationAuthorizationRequirement?

Reply:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authorization.iauthorizationrequirement?view=aspnetcore-2.2